capstone/src/main/java/com/google/sps/Objects/User.java

```diff
@@ -1,32 +1,25 @@
 package com.google.sps.Objects;
 
 import java.util.ArrayList;
+import java.util.LinkedHashSet;
 
 public final class User {
 
   private final String userId;
-  private final String name;
-  private final String firstName;
-  private final String lastName;
+  private String name;
+  private String firstName;
+  private String lastName;
   private final String email;
-  private final String phoneNumber;
-  private final ArrayList<Badge> badges;
-  private final ArrayList<String> groups;
+  private String phoneNumber;
+  private final LinkedHashSet<Badge> badges;
+  private final LinkedHashSet<Group> groups;
```

Comment:
Just to understand, is there a reason you chose LinkedHashSet instead of HashSet?

Reply:
because LinkedHashSet preserves the order and for something like badges I think it would be good to display them in the order/reverse order you received them

Reply:
Good reason! I'm unsure if datastore associates a timestamp with the Entities that are created/updated but I do remember y'all using a timestamp to sort the comments in your portfolio, so I recommend you do something similar here. If datastore has its own timestamp it's possible that it gets updated whenever that Entity is updated, if so I recommend you add a creation timestamp to the Badge object to remind y'all to add that when y'all create the Badges. 
Should Groups have the same ordering? If so, similarly, add a timestamp for that as well

Reply:
Hmm so should I change these to just a HashSet then?

Reply:
The issue I see with that is that even if we store a timestamp with the Badge or Group entities, and retrieve and add them to the User object in the desired order, by adding them to a HashSet we can no longer guarantee the order is preserved.

Reply:
Well I guess there are two things to consider here, Badges in the database and Badges outside of the database. If you can do similarly to what you did for the Comments for your portfolio where you were able to sort them by the timestamp, then a LinkedHashSet is likely not necessary because the data will already be sorted, though the LinkedHashSet would help maintain the order while you do things with the data outside of the database. Also I'm not sure if updating a Badge entity modifies the related timestamp (is the timestamp from datastore or is it an attribute of the Badge entity?). If the timestamp isn't modified then just do what you had done before with the Comments. If you feel like you'd need to maintain the order after you retrieve the sorted list from the database, then continue to use the LinkedHashSet, otherwise a simple HashSet works

Reply:
Sorry I didn't mean to confuse you between the Linked/HashSet. I think LinkedHashSet works and I agree with your comment, I was just asking about the timestamp because I don't know how the timestamp itself works.

Reply:
Oh, well I'm not sure how Anika and Lucy did it but the timestamp I used doesn't come from Datastore. I don't think Datastore stores a timestamp. I used System.currentTimeMillis() to get the time as a long.

Reply:
That should work for here as well! Just add a timestamp attribute to the Badge and Group objects, then I'll approve :-)